src/shared/utils/rateLimitDetector.ts

```
@@ -10,3 +10,181 @@ const RATE_LIMIT_SUBSTRING = "You've hit your limit";
+ * the parser bails out. Example: "reset at 3pm (PST) next week" — the naive
+ * "today or tomorrow" rollover would fire in hours instead of a week.
+ */
+const DAY_SHIFT_QUALIFIER_RE =
+  /\b(?:next\s+week|next\s+month|tomorrow|yesterday|on\s+(?:mon|tue|wed|thu|fri|sat|sun)[a-z]*)\b/i;
+
+// ---------------------------------------------------------------------------
+// Relative durations: "resets in 2 hours", "resets in 45 minutes"
+// ---------------------------------------------------------------------------
+
+const RELATIVE_RESET_RE =
+  /reset(?:s|ting)?\s+in\s+(?:about\s+|around\s+|~\s*)?(\d+(?:\.\d+)?)\s*(seconds?|secs?|minutes?|mins?|hours?|hrs?|h|m|s)\b/i;
```

ESLint

[error] Simplify this regular expression to reduce its complexity from 43 to the 20 allowed.

(sonarjs/regex-complexity)